cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/CheckedQueryException.java

```diff
@@ -0,0 +1,8 @@
+package com.cloudant.sync.query;
+
+/**
+ * Created by rhys on 04/07/2016.
+ */
+//FIXME: Give this a propername
```

Comment:
I guess you are planning to do this FIXME as part of the exception handling merge to the branch? I think something like `DatastoreQueryException` would be a good superclass for all of these query based exceptions. Probably better than just `QueryException` as that could be confused as SQL or even a server side query issue.